process/block/preprocess/transactions.go

```diff
@@ -537,23 +605,50 @@ func (txs *transactions) CreateAndProcessMiniBlock(
 			continue
 		}
 
-		currTxGasLimit := txs.economicsFee.ComputeGasLimit(orderedTxs[index])
-		if core.IsSmartContractAddress(orderedTxs[index].RcvAddr) {
-			currTxGasLimit = orderedTxs[index].GasLimit
+		gasConsumedByTxInSenderShard, gasConsumedByTxInReceiverShard, err := txs.gasHandler.ComputeGasConsumedByTx(
+			senderShardId,
+			receiverShardId,
+			orderedTxs[index])
+
+		if err != nil {
+			continue
+		}
+
+		var isMaxGasLimitReached bool
+		var gasConsumedInSelfShard uint64
+		if txs.shardCoordinator.SelfId() == miniBlock.SenderShardID {
```

Comment:
these conditions could be extracted in a method.

Reply:
extracted